### PR TITLE
Average Callback and Optimizers example 

### LINKED ISSUE #927: Add a tutorial for MovingAverage optimizer
We need a tutorial that shows how to use `MovingAverage` optimizer along with the custom callback for saving: `AverageModelCheckpoint`. Can alternatively use `StochasticAverage`. 

~~Currently blocked by #926~~

### PR DESCRIPTION
**Pull request by abhichou4**
Closes #927

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Can we vote to use 4 spaces to indent the notebooks? Let's start the revolution!

File: docs/tutorials/average_optimizers_callback.ipynb [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+        "id": "vc1eVXwjiFle",
+        "colab_type": "code",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 52
+        },
+        "outputId": "111f2d45-8276-4c73-90c7-c3c5a01b7603"
+      },
+      "source": [
+        "!pip install --upgrade tfa-nightly\n",
```
Comment (gabrieldemarmiesse) [inline docs/tutorials/average_optimizers_callback.ipynb]: Is it necessary to install the nightly or does it work with the 0.8.2 too?

File: docs/tutorials/average_optimizers_callback.ipynb [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+      },
+      "source": [
+        "#Callback \n",
+        "checkpoint_path = \"./training/cp-{epoch:04d}.ckpt\"\n",
+        "checkpoint_dir = os.path.dirname(checkpoint_path)\n",
+        "\n",
+        "cp_callback = tf.keras.callbacks.ModelCheckpoint(filepath=checkpoint_dir,\n",
+        "                                                 save_weights_only=True,\n",
+        "                                                 verbose=1)\n",
+        "avg_callback = tfa.callbacks.average_model_checkpoint.AverageModelCheckpoint(filepath=checkpoint_dir, \n",
```
Comment (gabrieldemarmiesse) [inline docs/tutorials/average_optimizers_callback.ipynb]: It seems this variable is never used. 

Could you also use the alias? `tfa.callbacks.AverageModelCheckpoint`

Comment (abhichou4) [reply docs/tutorials/average_optimizers_callback.ipynb]: I've added `load_weights` before evaluation to demonstrate the callback works fine.

File: docs/tutorials/average_optimizers_callback.ipynb (hunk 1 above)
Comment (abhichou4) [reply docs/tutorials/average_optimizers_callback.ipynb]: Yes, it was redundant. Removed it.

Comment (abhichou4) [reply docs/tutorials/average_optimizers_callback.ipynb]: `tfds` is also never used. Will remove that too.

File: docs/tutorials/average_optimizers_callback.ipynb [hunk 3]
```diff
@@ -0,0 +1,636 @@
...
+          "base_uri": "https://localhost:8080/",
+          "height": 471
+        }
+      },
+      "source": [
+        "#Build Model\n",
+        "model = create_model(moving_avg_sgd)\n",
+        "\n",
+        "#Train the network\n",
+        "model.fit(fmnist_train_ds, epochs=5, callbacks=[cp_callback])"
```
Comment (gabrieldemarmiesse) [inline docs/tutorials/average_optimizers_callback.ipynb]:
```suggestion
        "model.fit(fmnist_train_ds, epochs=5, callbacks=[avg_callback])"
```
Is that intended that the normal callback is used for `moving_avg_sgd` or am I missing something?

File: docs/tutorials/average_optimizers_callback.ipynb [hunk 4]
```diff
@@ -0,0 +1,636 @@
...
+        "colab_type": "code",
+        "outputId": "97fb462e-2c08-41d5-956d-a2ddb5140ee9",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 399
+        }
+      },
+      "source": [
+        "#Build Model\n",
+        "model.load_weights(checkpoint_dir)\n",
```
Comment (gabrieldemarmiesse) [inline docs/tutorials/average_optimizers_callback.ipynb]: This line does not seem to be useful.

File: docs/tutorials/average_optimizers_callback.ipynb [hunk 5]
```diff
@@ -0,0 +1,636 @@
...
+        "model = create_model(stocastic_avg_sgd)\n",
+        "\n",
+        "#Train the network\n",
+        "model.fit(fmnist_train_ds, epochs=5, callbacks=[cp_callback])"
```
Comment (gabrieldemarmiesse) [inline docs/tutorials/average_optimizers_callback.ipynb]:
```suggestion
        "model.fit(fmnist_train_ds, epochs=5, callbacks=[avg_callback])"
```

Same thing here. It's `stocastic_avg_sgd`, are you sure we shouldn't use `avg_callback` instead?

File: docs/tutorials/average_optimizers_callback.ipynb (hunk 3 above)
Comment (abhichou4) [reply docs/tutorials/average_optimizers_callback.ipynb]: No I missed it.

File: docs/tutorials/average_optimizers_callback.ipynb (hunk 4 above)
Comment (abhichou4) [reply docs/tutorials/average_optimizers_callback.ipynb]: Yes it's redundant. I think I'll switch to the colab workflow for better diffs.

File: docs/tutorials/average_optimizers_callback.ipynb (hunk 5 above)
Comment (abhichou4) [reply docs/tutorials/average_optimizers_callback.ipynb]: Missed it. Thanks!

**Pull request merged**